blackbox/docs/general/builtins/window-functions.rst

```diff
@@ -0,0 +1,85 @@
+.. highlight:: psql
+.. _window-functions:
+
+================
+Window Functions
+================
+
+.. rubric:: Table of Contents
+
+.. contents::
+   :local:
+
+Introduction
+============
+:ref:`Aggregation <aggregation>` functions can be executed over a subset of the
+rows returned by a query.
+Multiple rows might be connected or semantically related to the current row, in
+which case the business requirements might entail executing the aggregations
+on all related rows and emitting the result for the current row, as opposed to
+executing aggregations on the entire query result set and emitting the result
+just once (as it would happen by executing a plain 
+:ref:`aggregation function <aggregation>`).
+The set of related rows are defining the ``window`` the function will be
+executed against.
+
+Window functions enable an elegant way to express queries that include time
```

Comment:
I'm not sure if we want to include general reasonings on why X feature is good.

Reply:
I quite like it (especially as it highlights the alternative which would be expensive and complicated). Happy to remove it if you've a strong opinion

Reply:
Don't have a strong opinion here.
@mfussenegger what do you think?

Reply:
No strong opinion, but probably a bit too early since none of the stated use cases can be done with an empty over clause.